lms/djangoapps/courseware/views.py

```diff
@@ -1370,3 +1371,33 @@ def _track_successful_certificate_generation(user_id, course_id):  # pylint: dis
                 }
             }
         )
+
+
+@require_http_methods(["GET", "POST"])
+def render_xblock(request, usage_key_string):
+    """
+    Returns an HttpResponse with HTML content for the xBlock with the given usage_key.
+    The returned HTML is a chromeless rendering of the xBlock (excluding content of the containing courseware).
+    """
+    usage_key = UsageKey.from_string(usage_key_string)
+    usage_key = usage_key.replace(course_key=modulestore().fill_in_run(usage_key.course_key))
+    course_key = usage_key.course_key
```

Comment:
Ah, now I'm remembering more of this. The key is ambiguous in the case of old mongo, but one of the big limitations of old mongo is that we can't actually store more than one run. So it is in fact, always going to return the right thing. Got it.